solr/solrj/src/java/org/apache/solr/client/solrj/io/stream/StatsStream.java

```diff
@@ -129,51 +121,46 @@ public StatsStream(StreamExpression expression, StreamFactory factory) throws IO
       zkHost = ((StreamExpressionValue)zkHostExpression.getParameter()).getValue();
     }
 
-    // We've got all the required items
-    init(collectionName, params, metrics, zkHost);
-  }
+    /*
+    if(null == zkHost){
+      throw new IOException(String.format(Locale.ROOT,"invalid expression %s - zkHost not found for collection '%s'",expression,collectionName));
+    }
+    */
 
-  public String getCollection() {
-    return this.collection;
-  }
+    // metrics, optional - if not provided then why are you using this?
+    Metric[] metrics = new Metric[metricExpressions.size()];
+    for(int idx = 0; idx < metricExpressions.size(); ++idx){
+      metrics[idx] = factory.constructMetric(metricExpressions.get(idx));
+    }
 
-  private void init(String collection,
-                    SolrParams params,
-                    Metric[] metrics,
-                    String zkHost) throws IOException {
-    this.zkHost  = zkHost;
-    this.collection = collection;
-    this.metrics = metrics;
-    this.params = params;
+    // We've got all the required items
+    init(zkHost, collectionName, params, metrics);
   }
 
   @Override
   public StreamExpressionParameter toExpression(StreamFactory factory) throws IOException {
+    // functionName(collectionName, param1, param2, ..., paramN, sort="comp", sum(fieldA), avg(fieldB))
+
     // function name
     StreamExpression expression = new StreamExpression(factory.getFunctionName(this.getClass()));
+
     // collection
-    if(collection.indexOf(',') > -1) {
-      expression.addParameter("\""+collection+"\"");
-    } else {
-      expression.addParameter(collection);
-    }
+    expression.addParameter(collection);
 
     // parameters
-    ModifiableSolrParams tmpParams = new ModifiableSolrParams(params);
-
-    for (Entry<String, String[]> param : tmpParams.getMap().entrySet()) {
-      expression.addParameter(new StreamExpressionNamedParameter(param.getKey(),
-          String.join(",", param.getValue())));
+    ModifiableSolrParams mParams = new ModifiableSolrParams(params);
+    for (Entry<String, String[]> param : mParams.getMap().entrySet()) {
```

Comment:
The ".getMap().entrySet()" portion here is needless; a SolrParams is Iterable.  I'll file a PR to make this change comprehensively as I see it commonly in streaming expressions code.